setupTests.js

```diff
@@ -1,25 +1,35 @@
-import TestUtils from "./__tests__/test-utils";
+/* eslint-disable import/no-extraneous-dependencies,global-require */
+import TestUtils from './__tests__/test-utils';
```

Comment:
Why not remove this?

Reply:
These are _lint_ rules that I found it better to ignore for now, since fixing them could create confusion or take too long to investigate the proper way to implement them:

### [`import/no-extraneous-dependencies`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-extraneous-dependencies.md)
This accuses the `mock-socket` dependency to be not explicitly declared on the `package.json` dependencies. Looking at [the rules we follow from `airbnb-base`](https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb-base/rules/imports.js#L68-L97), I see that the setup filename we've chosen is not within their convention.

At a future time we could move the setup files into the `__tests__` folder and this would be solved.

### [`global-require`](https://eslint.org/docs/rules/global-require)
On lines `18` and `29` we require javascript files within a very controlled clojure scope on Jest. If we imported these files into global variables and tried to use them in the mock `factory` method, we would get the following error:
```
babel-plugin-jest-hoist: The module factory of `jest.mock()` is not allowed to reference any out-of-scope variables.
```